apps/site/pages/en/learn/diagnostics/memory/understanding-and-tuning-memory.md

```diff
@@ -0,0 +1,137 @@
+
+### `--max-old-space-size`
+
+This flag sets a limit on the size of the **Old Space** in the V8 heap, where long-lived objects are stored. If your application uses a significant amount of memory, you might need to adjust this limit.
+
+To increase the Old Space limit to 4 GB, for example, use:
+
+```bash
+node --max-old-space-size=4096 app.js
+```
+
+This sets the Old Space size to 4096 MB (4 GB), but you should adjust this based on your system's available memory.
+
+### `--max-semi-space-size`
+
+The `--max-semi-space-size` flag controls the size of the **New Space** in the V8 heap. New Space stores newly created objects, which are garbage collected frequently. Increasing this size can help reduce the frequency of minor garbage collections, which can in turn improve performance if your application allocates a lot of short-lived objects.
+
+For example:
+
+```bash
+node --max-semi-space-size=512 app.js
+```
+
+This increases the New Space to 512 MB (compared to the default, which is typically much smaller). This adjustment can help in high-throughput applications where frequent garbage collection of short-lived objects is adding noticeable overhead.
+
+### `--gc-interval`
+
+This flag adjusts how frequently garbage collection cycles occur. V8 normally determines this interval automatically based on heuristics, but in special cases, you might want to override it.
+
+For example:
+
+```bash
+node --gc-interval=100 app.js
+```
+
+Setting this to 100 forces V8 to attempt garbage collection every 100 ms. This might be useful in low-latency systems where predictable memory cleanup is needed, such as a long-running service that must consistently stay within a narrow memory footprint. However, changing this without careful testing can degrade performance due to excessive garbage collection.
+
+### `--expose-gc`
+
+With the `--expose-gc` flag, you can manually trigger garbage collection from your application code. This can be helpful in specific cases where you know a large chunk of memory is no longer needed and want to release it immediately—for example, after processing a large data set or completing a batch job.
+
+To enable it, start your app with:
+
+```bash
+node --expose-gc app.js
+```
+
+Then, within your code, you can call:
+
+```javascript
+global.gc();
+```
+
+Keep in mind that manually triggering garbage collection **does not disable** the normal GC algorithm. V8 will still perform automatic garbage collection as needed. Manual calls are supplemental and should be used with caution, as overuse can negatively impact performance.
+
+## Putting It All Together
+
+By adjusting settings for the Old Space and New Space sizes, selectively triggering garbage collection, and configuring heap limits, you can optimize your application’s memory usage and improve its overall performance. These tools give you the power to better manage memory in high-demand scenarios and maintain stability as your applications scale.
```

Comment:
I think it would be good to add some references to the V8 blog:

https://v8.dev/blog/trash-talk
https://v8.dev/blog/orinoco-parallel-scavenger